vita3k/patch/src/patch.cpp

```diff
@@ -0,0 +1,81 @@
+// Vita3K emulator project
+// Copyright (C) 2024 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include "patch/patch.h"
+
+#include <util/log.h>
+
+std::vector<Patch> get_patches(fs::path &path, const std::string &titleid) {
+    // Find a file in the path with the titleid
+    std::vector<Patch> patches;
+
+    LOG_INFO("Looking for patches for titleid {}", titleid);
+
+    for (auto &entry : fs::directory_iterator(path)) {
+        auto filename = entry.path().filename().string();
+        // Just in case users decide to use lowercase filenames
+        auto upper_filename = std::transform(filename.begin(), filename.end(), filename.begin(), ::toupper);
```

Comment:
IMO, the filenames of patches should be case sensitive just in case theres 2 files with different casing but for the same titleid and the user is dumb and cant figure out why their patches arent working. Also just a bit for simplification and make every patch's titleid be uppercase c:

I want to know what the others think of this tho

Reply:
My concern is that there is no way for the user to know that their patch isn't being read simply because it is in lower case. I think it's better to assume the user might be dumb and will used lower/mixed case names because there is no real reason for them to be case-sensitive anyways.

I can already imagine an issue being submitted with the title "Patches not applying", with the answer to their problem being "make the filename uppercase", which some might annoying. Or imagine someone creates a lowercase patch file, tries it, sees that it doesn't work, and then just goes "ugh, I guess the patch system doesn't work". This kind of thing is easily preventable by keeping this line, but I'd also be happy to hear what others think :)

Reply:
I think case-insensitive implementation would be better.

1. It's difficult for end-users using Windows systems to understand that software behavior can differ due to filename case differences.
2. Even when multiple patches with only case differences exist on Linux systems, all those patches are applied regardless of case.

Among these two reasons, reason 1 is more important. I'm also curious about others' opinions on this issue.

Reply:
I also agree that it should be case-insensitive. I don't know of any case where this could be a problem.

Reply:
> 1. It's difficult for end-users using Windows systems to understand that software behavior can differ due to filename case differences.

Windows wouldnt be affected by the change as the filesystem is case-insesitive already, it would only change for linux/android users, but i see your point, it seems that i underestimated how utterly dumb the user can be especially android